assets/js/calypsoify.js

```diff
@@ -15,33 +15,42 @@
      * Record checklist task click
      */
     $( '.checklist__task-title a, .checklist__task-secondary a' ).click( function() {
-        var $task = $( this ).closest( '.checklist__task' )
-        var status = $task.hasClass( 'is-completed' ) ? 'complete' : 'incomplete';
-        var taskId = $task.data('id');
-        var taskTitle = $task.data('title');
+        const $task = $( this ).closest( '.checklist__task' )
+        const status = $task.hasClass( 'is-completed' ) ? 'complete' : 'incomplete';
+        const taskId = $task.data('id');
+        const taskTitle = $task.data('title');
+        const href = $( this ).attr('href');
 
-        window.jpTracksAJAX.record_ajax_event(
-            'atomic_wc_tasklist_click',
-            'click',
-            {
-                id: taskId,
-                title: taskTitle, 
-                status: status,
-            }
-        );
+        if ( window.jpTracksAJAX ) {
+            const trackedEvent = window.jpTracksAJAX.record_ajax_event(
+                'atomic_wc_tasklist_click',
+                'click',
+                {
+                    id: taskId,
+                    title: taskTitle, 
+                    status: status,
+                }
+            );
+            trackedEvent.complete( function() {
+                window.location = href;
+            } );
+        } else {
+            window.location = href;
+        }
     } );
 
     /**
      * Track 'I'm done' completion on task list
      */
     $( '.setup-footer a' ).click( function(e) {
         e.preventDefault();
-        var progressNumber = $( '.checklist__header-progress-number' ).text().split( '/' );
-        var complete = progressNumber[0];
-        var total = progressNumber[1];
-        var percentage = parseFloat( complete / total ).toFixed( 2 ) * 100;
-        if (window.jpTracksAJAX) {
-            window.jpTracksAJAX.record_ajax_event(
+        const progressNumber = $( '.checklist__header-progress-number' ).text().split( '/' );
```

Comment:
There is a slight delay now while the request completes. Should we consider adding a `disabled` class to the element to have it verify the button has indeed been clicked and the "form" is submitting?

![image](https://user-images.githubusercontent.com/22080/48848181-2c3e8580-ed58-11e8-9159-b5c46d2f5319.png)

Reply:
I think that's fine.